src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/EmbeddingProvidersConfig.java

```diff
@@ -189,28 +189,49 @@ public String toString() {
     interface RequestProperties {
 
       /**
-       * The maximum number of retries to attempt before failing the request.
+       * Specifies the maximum number of attempts before failing. Default is 3 (1 request + 2
+       * retries).
        *
-       * @return The maximum number of retries to attempt before failing the request.
+       * @return The maximum number of attempts before failing.
        */
       @WithDefault("3")
-      int maxRetries();
+      int atMostRetries();
 
       /**
-       * The back off time between retries in milliseconds.
+       * The initial delay between retries in milliseconds. The first retry occurs after the
+       * specified delay (default 100 ms), doubling each time until reaching maxBackOffMillis.
        *
-       * @return The delay between retries in milliseconds.
+       * @return The initial delay between retries in milliseconds.
        */
       @WithDefault("100")
-      int retryDelayMillis();
+      int initialBackOffMillis();
 
       /**
-       * The timeout for the request in milliseconds.
+       * The maximum duration for a request to complete in milliseconds. Default is 5000 ms (5
```

Comment:
nit: either this description is wrong, or name of setting is misleading: either this is the maximum total duration OR it is timeout for read from connection (socket).